### PR TITLE
infinite scrolling for courses-users chat

### PR DESCRIPTION
**Pull request by nguyenngocha**

### CODE REVIEW

File: app/views/layouts/_chat_header.html.erb [hunk 1]
```diff
@@ -35,7 +35,19 @@
   <aside class="main-sidebar-chat">
     <section class="sidebar" id="chat-sidebar">
       <ul class="sidebar-menu">
-        <%= render "shared/user_chat" %>
+        <div id="courses">
+          <% @courses.each do |course|%>
```
Comment (yongsokheng) [inline app/views/layouts/_chat_header.html.erb]: convention

File: app/assets/javascripts/pagination.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+$(document).on('turbolinks:load', function() {
+  if ($('#infinite-scrolling').size() > 0) {
+    $('#chat-sidebar').on('scroll', function() {
+      var more_posts_url;
+      more_posts_url = $('.pagination .next a').attr('href');
+      if (more_posts_url && $(window).scrollTop() > $(document).height() - $(window).height() - 60) {
+        $('.pagination').html('<img src="/assets/ajax-loader.gif" alt="Loading..." title="Loading..." />');
```
Comment (yongsokheng) [inline app/assets/javascripts/pagination.js]: i18n va dung '' trong " " nha

File: app/views/chats/index.js.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+$("#users").append("<%= j render partial: 'shared/user_chat', collection: @users, as: :user %>")
+<% if @users.next_page %>
+  $(".pagination").replaceWith("<%= j paginate @users %>");
+<% else %>
+  $(window).off("scroll");
```
Comment (yongsokheng) [inline app/views/chats/index.js.erb]: $('#chat-sidebar').off("scroll");

File: app/controllers/chats_controller.rb [hunk 4]
```diff
@@ -1,4 +1,11 @@
 class ChatsController < ApplicationController
   def index
+    @courses = current_user.courses
+    @users = User.all.per_page_kaminari(params[:page]).per Settings.per_page
```
Comment (yongsokheng) [inline app/controllers/chats_controller.rb]: bo all di co chay ko

File: config/locales/chat.en.yml [hunk 5]
```diff
@@ -0,0 +1,2 @@
+en:
+  load: "Loading..."
```
Comment (ngtrduc) [inline config/locales/chat.en.yml]: end file

File: app/views/layouts/_chat_header.html.erb [hunk 6]
```diff
@@ -35,7 +35,19 @@
   <aside class="main-sidebar-chat">
     <section class="sidebar" id="chat-sidebar">
       <ul class="sidebar-menu">
-        <%= render "shared/user_chat" %>
+        <div id="courses">
+          <%= render partial: "shared/chat_room", collection: @courses, as: :object %>
```
Comment (yongsokheng) [inline app/views/layouts/_chat_header.html.erb]: doi ten object thanh chat_room thi de hieu hon

File: config/locales/chat.en.yml [hunk 7]
```diff
@@ -0,0 +1,2 @@
+en:
+  loading: "Loading..."
```
Comment (yongsokheng) [inline config/locales/chat.en.yml]: en: 
   chats:
        loading: "Loading ...."

**Pull request merged**